Analysis/IsolatedLeptonFinder/src/IsolatedLeptonFinderProcessor.cc

```diff
@@ -280,38 +324,212 @@ void IsolatedLeptonFinderProcessor::processEvent( LCEvent * evt ) {
 		}
 	}
 
-	// PFO loop
+
+	// Undressed leptons
+	std::vector<int> goodLeptonIndices;
 	int npfo = _pfoCol->getNumberOfElements();
 	for (int i = 0; i < npfo; i++ ) {
 		ReconstructedParticle* pfo = dynamic_cast<ReconstructedParticle*>( _pfoCol->getElementAt(i) );
 
-		if ( IsIsolatedLepton( pfo ) ) 
+		if ( !IsGoodLepton( pfo )){
+			otPFOsRemovedIsoLepCol->addElement( pfo );
+			continue;
+		}
+
+		if  ( IsIsolatedLepton( pfo, false ) ){
+			streamlog_out(DEBUG) << "ISOLATION undressed "<<pfo->getType()<<": SUCCESS !" << std::endl;
 			otIsoLepCol->addElement( pfo );
-		else 
+		}
+		else{
+			streamlog_out(DEBUG) << "ISOLATION undressed "<<pfo->getType()<<": FAILED" << std::endl;
 			otPFOsRemovedIsoLepCol->addElement( pfo );
+		}
+
+		// remember lepton indices for dressing
+		if (IsLepton( pfo )) goodLeptonIndices.push_back(i);
 	}
 
-	streamlog_out(DEBUG) << "   processing event: " << evt->getEventNumber() 
-		<< "   in run:  " << evt->getRunNumber() 
-		<< std::endl ;
 
-	// Add PFOs to new collection
-	evt->addCollection( otPFOsRemovedIsoLepCol, _outputPFOsRemovedIsoLepCollection.c_str() );
-	evt->addCollection( otIsoLepCol, _outputIsoLepCollection.c_str() );
-}
+	// Dressed leptons
+	// order by energy
+	for (unsigned int i = 1; i < goodLeptonIndices.size(); ++i)
```

Comment:
use `std::sort` to do the ordering

Reply:
Thanks, I was looking for a solution with my lambda function and sort but gave up at some point. The `not` is not necessary, it needs a greater comparison.

Reply:
So this is done as well!